js/ui/grid_core/ui.grid_core.keyboard_navigation.js

```diff
@@ -1106,10 +1106,7 @@ var KeyboardNavigationController = core.ViewController.inherit({
 
     _ctrlFKeyHandler: function(eventArgs) {
         if(eventArgs.ctrl && this.option("searchPanel") && this.option("searchPanel").visible) {
-            ///#DEBUG
-            this._testHeaderPanelFocused = true;
-            ///#ENDDEBUG
-            this._headerPanel.focus();
+            this._headerPanel._getSearchTextEditor().focus();
```

Comment:
The _getSearchTextEditor method can return null. Also please rename _getSearchTextEditor to getSearchTextEditor because you call this method from keyboardNavigation and it is declared in searchPanel.

Reply:
Replace this.option("searchPanel") && this.option("searchPanel").visible 
to
this.option("searchPanel.visible")